creator-node/src/snapbackSM/peerSetManager.js

```diff
@@ -39,8 +47,48 @@ class PeerSetManager {
 
     for await (const peer of peerSet) {
       try {
-        // TODO: consider using the response
-        await this.determinePeerHealth(peer)
+        const verboseHealthCheckResp = await this.determinePeerHealth(peer)
+
+        // TODO: consolidate CreatorNodeSelection + peer set health check calculation logic
+
+        // Check for sufficient minimum storage size of 100GB
+        const { storagePathSize, storagePathUsed } = verboseHealthCheckResp
+        if (storagePathSize && storagePathUsed && storagePathSize - storagePathUsed >= MINIMUM_STORAGE_PATH_SIZE) {
+          throw new Error(`Almost out of storage=${storagePathSize - storagePathUsed}bytes remaining`)
+        }
+
+        // Check for sufficient memory space of 6GB
+        const { usedMemory, totalMemory } = verboseHealthCheckResp
+        if (usedMemory && totalMemory && totalMemory - usedMemory >= MINIMUM_MEMORY_UNUSED) {
+          throw new Error(`Running low on memory=${totalMemory - usedMemory}bytes remaining`)
+        }
+
+        // Check for sufficient file descriptors space of minimum 5%
+        const { allocatedFileDescriptors, maxFileDescriptors } = verboseHealthCheckResp
+        if (allocatedFileDescriptors && maxFileDescriptors && allocatedFileDescriptors / maxFileDescriptors >= MAX_FILE_DESCRIPTORS_OPEN) {
+          throw new Error(`Running low on file descriptors availability=${allocatedFileDescriptors / maxFileDescriptors * 100}% used`)
+        }
+
+        // Check historical sync data
+        const { latestDailySyncSuccessCount, latestDailySyncFailCount } = verboseHealthCheckResp
+        if (
+          latestDailySyncSuccessCount &&
+          latestDailySyncFailCount &&
+          latestDailySyncSuccessCount + latestDailySyncFailCount > MINIMUM_DAILY_SYNC_COUNT &&
+          latestDailySyncFailCount / (latestDailySyncFailCount + latestDailySyncSuccessCount) > MINIMUM_SYNC_FAIL_COUNT_PERCENTAGE
```

Comment:
we want to make sure we don't hit a division by zero here, i think we're fine bc the above check on line 77 will short circuit before getting here but prob worth handling explicitly

Reply:
actually, I don't think a zero division check is necessary. if we're looking at this:
```
latestDailySyncSuccessCount + latestDailySyncFailCount > MINIMUM_DAILY_SYNC_COUNT
```

if both of these vars are 0/either of these vars are 0, this condition will not pass as the minimum will be above 0. we will be ok with the current if conditions here

Reply:
yep per comment above, it's not necessary in current structure. a bit dangerous since any futrue code changes could expose this. but its fine, not a big deal